### PR TITLE
add a blkproto.mli with only the needed definitions

### PR DESCRIPTION
**Pull request by yomimono**

### CODE REVIEW

File: lib/blkproto.mli [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    media: Media.t;
+    removable: bool;
+  }
+
+  val to_assoc_list : t -> (int * (string * string)) list
+end
+
+module Req : sig
+  type seg = {
+    gref : int32;
```
Comment (talex5) [inline lib/blkproto.mli]: Maybe replace all `int32`s with `Gntref.t`, once #79 is merged?

File: lib/blkproto.mli [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    sectors : int64;
+    media : Media.t;
+    mode : Mode.t;
+  }
+
+  val to_assoc_list : t -> (string * string) list
+end
+
+module Hotplug : sig
+  val _hotplug_status : string
```
Comment (talex5) [inline lib/blkproto.mli]: Is the `_` here intentional?

File: lib/blkproto.ml [hunk 3]
```diff
@@ -37,7 +37,7 @@ module Mode = struct
   let to_string = function
     | ReadOnly -> "r"
     | ReadWrite -> "w"
-  let of_string = function
+  let _of_string = function
```
Comment (talex5) [inline lib/blkproto.ml]: Are these functions unused? I guess we should either delete them or expose them in the API.

File: lib/blkproto.ml [hunk 4]
```diff
@@ -260,6 +260,7 @@ module Req = struct
   | Flush -> "Flush" | Op_reserved_1 -> "Op_reserved_1" | Trim -> "Trim"
   | Indirect_op -> "Indirect_op"
 
+  [@@@warning "-38"]
   exception Unknown_request_type of int
```
Comment (talex5) [inline lib/blkproto.ml]: Just delete it? Also, I think `@@@` applies to everything following, not just this item, which from the formatting I'm guessing isn't what you intended.

File: lib/blkproto.mli (hunk 2 above)
Comment (yomimono) [reply lib/blkproto.mli]: Yes, this is the name given in `blkproto.ml`.  Your guess is as good as mine.

File: lib/blkproto.ml (hunk 3 above)
Comment (yomimono) [reply lib/blkproto.ml]: I'd lean toward deleting them, personally.

File: lib/blkproto.mli (hunk 1 above)
Comment (yomimono) [reply lib/blkproto.mli]: Yep, that's what I'd like to do, but I wanted #79 to be considered separately from this maintenance/quality-of-life stuff.

File: lib/blkproto.mli (hunk 2 above)
Comment (djs55) [reply lib/blkproto.mli]: Probably some strange naming convention for constant strings before anyone realised that `_` had special meaning?

File: lib/blkproto.ml (hunk 3 above)
Comment (djs55) [reply lib/blkproto.ml]: Deleting them seems good. Can be a separate PR though since this one is only about exposing the minimal mli

**Comment by djs55** [comment]
Looks like the code still uses an old polymorphic variant result rather than the new (!) result type
```
# File "lib/blkback.ml", line 448, characters 8-13:
# Error: This pattern matches values of type [? `OK of 'a ]
#        but a pattern was expected which matches values of type
#          (Blkproto.RingInfo.t, [ `Msg of string ]) Stdlib.result =
#            (Blkproto.RingInfo.t, [ `Msg of string ]) Pervasives.result
```

**Comment by yomimono** [comment]
I've rebased (to solve the polyvar/result mismatch) and removed a bunch of apparently-unused stuff.

**Pull request merged**